Source/DafnyDriver/DafnyDriver.cs

```diff
@@ -469,7 +469,7 @@ static string WriteDafnyProgramToFiles(string dafnyProgramName, string targetPro
           throw new cce.UnreachableException();
       }
       string targetBaseName = Path.ChangeExtension(baseName, targetExtension);
-      string targetDir = Path.Combine(Path.GetDirectoryName(dafnyProgramName), targetBaseDir);
+      string targetDir = DafnyOptions.O.JavaCompileTargetDirectory ?? Path.Combine(Path.GetDirectoryName(dafnyProgramName), targetBaseDir);
```

Comment:
The Go backend also uses a target directory. It appends `-go` to the `dafnyProgramName` when constructing the target base directory (line 461 of DafnyDriver.cs). This is nice when you want to, for example, find and delete all such directories on your disk. Can we do the same (but appending `-java`) for the Java backend? (Either here or a separate PR.)